test/Microsoft.ML.Tests/AnomalyDetectionTests.cs

```diff
@@ -29,33 +22,57 @@ public AnomalyDetectionTests(ITestOutputHelper output) : base(output)
         [Fact]
         public void RandomizedPcaTrainerBaselineTest()
         {
-            var mlContext = new MLContext(seed: 1, conc: 1);
-            string featureColumn = "NumericFeatures";
-
             var reader = new TextLoader(Env, new TextLoader.Options()
             {
-                HasHeader = true,
+                HasHeader = false,
                 Separator = "\t",
                 Columns = new[]
                 {
                     new TextLoader.Column("Label", DataKind.R4, 0),
-                    new TextLoader.Column(featureColumn, DataKind.R4, new [] { new TextLoader.Range(1, 784) })
+                    new TextLoader.Column(DefaultColumnNames.Features, DataKind.R4, new [] { new TextLoader.Range(1, 784) })
                 }
             });
 
             var trainData = reader.Read(GetDataPath(TestDatasets.mnistOneClass.trainFilename));
             var testData = reader.Read(GetDataPath(TestDatasets.mnistOneClass.testFilename));
 
-            var pipeline = ML.AnomalyDetection.Trainers.RandomizedPca(featureColumn);
+            var pipeline = ML.AnomalyDetection.Trainers.RandomizedPca();
 
             var transformer = pipeline.Fit(trainData);
             var transformedData = transformer.Transform(testData);
 
             // Evaluate
             var metrics = ML.AnomalyDetection.Evaluate(transformedData, k: 5);
 
-            Assert.Equal(0.98269, metrics.Auc, 5);
+            Assert.Equal(0.98286, metrics.Auc, 5);
             Assert.Equal(0.90000, metrics.DrAtK, 5);
         }
+
+        /// <summary>
+        /// Test anomaly detection when the test data has no anomalies
+        /// </summary>
+        [Fact]
+        public void NoAnomalyTest()
+        {
+            var reader = new TextLoader(Env, new TextLoader.Options()
```

Comment:
>TextLoader [](start = 29, length = 10)

duplicate code with method above. Refactor. Perhaps make this a method off the dataset.mnistOneClass?

Reply:
Moved it into private method